pkg/daemon/ceph/agent/flexvolume/controller.go

```diff
@@ -301,12 +302,35 @@ func (c *Controller) GetAttachInfoFromMountDir(mountDir string, attachOptions *A
 			return fmt.Errorf("failed to get pods in namespace %s: %+v", attachOptions.PodNamespace, err)
 		}
 
-		pod := findPodByID(pods, types.UID(attachOptions.PodID))
+		pod = findPodByID(pods, types.UID(attachOptions.PodID))
 		if pod != nil {
 			attachOptions.Pod = pod.GetName()
 		}
 	}
 
+	if attachOptions.PodUser == 0 || attachOptions.PodGroup == 0 {
+		if pod == nil {
+			pod, err = c.context.Clientset.CoreV1().Pods(attachOptions.PodNamespace).Get(attachOptions.Pod, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to get pod %s in namespace %s: %+v", attachOptions.Pod, attachOptions.PodNamespace, err)
+			}
+		}
+		if pod != nil && pod.Spec.SecurityContext != nil {
+			if attachOptions.PodUser == 0 && pod.Spec.SecurityContext.RunAsUser != nil {
+				attachOptions.PodUser = *pod.Spec.SecurityContext.RunAsUser
+				if pod.Spec.SecurityContext.FSGroup == nil {
+					attachOptions.PodGroup = attachOptions.PodUser
+				}
+			}
+			if attachOptions.PodGroup == 0 && pod.Spec.SecurityContext.FSGroup != nil {
+				attachOptions.PodGroup = *pod.Spec.SecurityContext.FSGroup
+				if pod.Spec.SecurityContext.RunAsUser == nil {
+					attachOptions.PodUser = attachOptions.PodGroup
```

Comment:
the code here is still necessary, right? I'm assuming your comment doesn't apply anymore since adding chown.

Reply:
I mean defaulting the uid to the gid if the uid isn't specified. This is necessary because the mounted volume isn't group writable and the uid isn't usually set in the pod scc. We could go further and look at the container scc, but I think it would make more sense to just mount the volume group writable (which means that we don't need to default the uid to gid).